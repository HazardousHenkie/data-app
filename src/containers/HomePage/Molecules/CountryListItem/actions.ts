import { action } from 'typesafe-actions'

import ActionTypes from './constants'

import CountryState from './types'

const setSelectedCountry = (country: CountryState['country']) =>
    action(ActionTypes.SET_SELECTED_COUNTRY, country)

export default setSelectedCountry
