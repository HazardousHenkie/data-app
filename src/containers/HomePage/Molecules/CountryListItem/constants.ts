import CountryState from './types'

enum ActionTypes {
    SET_SELECTED_COUNTRY = 'components/CountriesList/SET_SELECTED_COUNTRY'
}

export const CountryItem: CountryState = {
    country: {
        alpha2Code: '',
        name: '',
        nativeName: '',
        capital: '',
        region: '',
        subregion: '',
        flag: '',
        currency: '',
        population: 0,
        latlng: [0, 0],
        currencies: [{ string: '' }],
        languages: [{ string: '' }],
        translations: { string: '' }
    }
}

export default ActionTypes
