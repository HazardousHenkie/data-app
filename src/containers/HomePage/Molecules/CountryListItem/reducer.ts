import { ContainerCountryState, ContainerCountryActions } from './types'

import ActionTypes from './constants'

export const initialCountryState: ContainerCountryState = {
    country: {}
}

function countriesListReducer(
    state: ContainerCountryState = initialCountryState,
    action: ContainerCountryActions
): ContainerCountryState {
    switch (action.type) {
        case ActionTypes.SET_SELECTED_COUNTRY:
            return {
                country: action.payload
            }
        default:
            return state
    }
}

export default countriesListReducer
