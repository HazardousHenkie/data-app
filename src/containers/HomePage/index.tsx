import React from 'react'

import { Helmet } from 'react-helmet'
import { useTranslation } from 'react-i18next'

import Drawer from 'components/Organisms/Drawer'

import OSMap from './map'

const HomePage: React.FC = () => {
    const { t } = useTranslation('homePage')

    return (
        <>
            <Helmet>
                <title>{t('homePage:title', 'Home Page')}</title>
                <meta
                    name={t('homePage:title', 'Home Page')}
                    content={t('homePage:description', 'Homepage description')}
                />
            </Helmet>

            <OSMap />

            <Drawer>
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttesttest
                testtesttesttesttesttesttesttesttesttesttest
            </Drawer>
        </>
    )
}

export default HomePage
