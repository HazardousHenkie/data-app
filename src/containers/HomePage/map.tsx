import React, { useState, useEffect, useRef } from 'react'

import { Map, MapProps, TileLayer } from 'react-leaflet'
import * as Leaflet from 'leaflet'

import Loader from 'components/Atoms/Loader'
import StyledMap from './styledComponents'

const OSMap: React.FC = () => {
    const [mapState, setMapState] = useState({ lat: 0.0, lng: 0.0 })
    const [zoom] = useState(6)
    const [loading, setLoading] = useState(true)
    const mapRef = useRef() as React.RefObject<Map<MapProps, Leaflet.Map>>

    useEffect(() => {
        const map = mapRef.current

        if (map) {
            map.leafletElement.locate()
        }
    }, [mapRef])

    const getLocationSuccess = (e: Leaflet.LocationEvent): void => {
        setMapState(e.latlng)
        setLoading(false)
    }

    const getLocationError = (): void => {
        setLoading(false)
    }

    return (
        <>
            {loading && <Loader />}

            <StyledMap
                zoomControl={false}
                center={[mapState.lat, mapState.lng]}
                onlocationfound={getLocationSuccess}
                onlocationerror={getLocationError}
                zoom={zoom}
                ref={mapRef}
            >
                <TileLayer
                    attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
                    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                />
            </StyledMap>
        </>
    )
}

export default OSMap
