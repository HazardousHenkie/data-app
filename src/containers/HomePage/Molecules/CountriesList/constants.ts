import { CountryItem } from 'containers/HomePage/Molecules/CountryListItem/constants'
import CountriesListState from './types'

enum ActionTypes {
    GET_COUNTRIES_DATA = 'components/CountriesList/GET_COUNTRIES_DATA',
    GET_COUNTRIES_DATA_SUCCESS = 'components/CountriesList/GET_COUNTRIES_DATA_SUCCESS',
    GET_COUNTRIES_DATA_ERROR = 'components/CountriesList/GET_COUNTRIES_DATA_ERROR'
}

export const initialCountriesHeaderState: CountriesListState = {
    error: false,
    loading: false,
    data: [CountryItem.country]
}

export default ActionTypes
