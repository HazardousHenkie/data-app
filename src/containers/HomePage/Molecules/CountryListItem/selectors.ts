import { createSelector, Selector } from 'reselect'

import { ApplicationRootState } from 'types'
import { CountryInterface } from './types'
import { CountryItem } from './constants'

export const initialCountryState = CountryItem

export const selectCountry = (state: ApplicationRootState) => {
    return state.country || initialCountryState
}

const makeSelectCountry = () =>
    createSelector(selectCountry, subState => subState.country) as Selector<
        unknown,
        CountryInterface
    >

export default makeSelectCountry
