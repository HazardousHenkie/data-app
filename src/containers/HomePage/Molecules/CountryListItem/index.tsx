import React from 'react'

import { Reducer } from 'redux'

import { useInjectReducer } from 'utils/redux-injectors'

import ListItem from '@material-ui/core/ListItem'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemIcon from '@material-ui/core/ListItemIcon'

import { useTranslation } from 'react-i18next'

import { useDispatch, useSelector } from 'react-redux'
import { createSelector } from 'reselect'
import { makeSelectData } from 'containers/HomePage/Molecules/CountriesList/selectors'
import { CountryInterface } from './types'
import setSelectedCountry from './actions'

import reducer from './reducer'

interface CountriesListItemProps {
    setOpen: React.Dispatch<React.SetStateAction<boolean>>
    listCountry: CountryInterface
}

const key = 'country'

const stateSelector = createSelector(makeSelectData(), countries => ({
    countries
}))

const CountriesListItem: React.FC<CountriesListItemProps> = ({
    setOpen,
    listCountry
}) => {
    const dispatch = useDispatch()
    const { i18n } = useTranslation()
    const { countries } = useSelector(stateSelector)

    useInjectReducer({ key, reducer: reducer as Reducer })

    const onClickCountry = (selectedName: string) => {
        const selectedCountry = countries.find(
            country => country.name === selectedName
        )

        if (selectedCountry) {
            setOpen(false)
            dispatch(setSelectedCountry(selectedCountry))
        }
    }

    return (
        <ListItem
            data-testid="CountriesListItem"
            button
            onClick={() => onClickCountry(listCountry.name)}
        >
            <ListItemIcon data-testid="CountriesListItemIcon">
                <img
                    width="30"
                    height="20"
                    src={listCountry.flag}
                    alt={listCountry.name}
                />
            </ListItemIcon>
            <ListItemText
                data-testid="CountriesListItemText"
                primary={
                    i18n.language === 'en'
                        ? listCountry.name
                        : listCountry.translations[i18n.language]
                }
            />
        </ListItem>
    )
}

export default CountriesListItem
