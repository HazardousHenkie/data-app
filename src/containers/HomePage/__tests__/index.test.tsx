import React from 'react'
import { render, act } from 'utils/test-utils'

import { CountryItem } from 'containers/HomePage/Molecules/CountryListItem/constants'
import setSelectedCountry from 'containers/HomePage/Molecules/CountryListItem/actions'

import { Helmet, HelmetTags, HelmetData } from 'react-helmet'

import mockFetch, { mockFetchCleanUp } from 'utils/request-test-utils'
import configureStore from 'store/configureStore'

import HomePage from '../index'

describe('<HomePage />', () => {
    let store = configureStore({})

    beforeEach(() => {
        mockFetch({})
    })

    afterEach(() => {
        store = configureStore({})
        mockFetchCleanUp()
    })

    it('should render like snapshot', () => {
        const component = render(<HomePage />)

        expect(component).toMatchSnapshot()
    })

    test('Check if translation text is shown', () => {
        render(<HomePage />)

        const helmet = Helmet.peek()
        expect(helmet.title).toEqual('Home Page')
    })

    it('should set metaDescription', () => {
        render(<HomePage />)

        const metaTags = [
            { name: 'Home Page', content: 'Homepage description' }
        ]

        const helmet = Helmet.peek()
        expect((helmet as HelmetTags & HelmetData).metaTags).toEqual(metaTags)
    })

    it('should render HeaderSearch', () => {
        const { getByTestId } = render(<HomePage />)

        expect(getByTestId('TopBar')).toBeInTheDocument()
    })

    it('should render OSMap', () => {
        const { getByTestId } = render(<HomePage />)

        expect(getByTestId('OSMap')).toBeInTheDocument()
    })

    it('should render drawer when country is set', () => {
        const { getByTestId } = render(<HomePage />, { store })

        act(() => {
            store.dispatch(
                setSelectedCountry({ ...CountryItem.country, alpha2Code: 'NL' })
            )
        })

        expect(getByTestId('drawer')).toBeInTheDocument()
    })
})
