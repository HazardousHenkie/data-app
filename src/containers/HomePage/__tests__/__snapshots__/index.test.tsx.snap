// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render like snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="HomePage"
      >
        <div
          class="sc-fzoLag SLPgS"
          data-testid="TopBar"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root sc-fznyAO cWfjdE MuiIconButton-colorInherit MuiIconButton-edgeStart"
            data-testid="IconButtonStyled"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="sc-fzoXzr IxHii"
            data-testid="SearchFieldWrapper"
            role="button"
            tabindex="0"
          >
            <form>
              <div
                class="MuiFormControl-root MuiTextField-root sc-fzozJi hWuMcr sc-fzoLsD lgrigy"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    placeholder="search"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-2"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          class="sc-AxmLO fGDwgL"
          data-testid="CountriesList"
          style="opacity: 0; visibility: hidden;"
        >
          <div
            class="sc-Axmtr kQOpxN"
            data-testid="inlineLoader"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="sc-fzpmMD bMrmBP leaflet-container leaflet-grab leaflet-touch-drag"
          style="position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            >
              <div
                class="leaflet-layer "
                style="z-index: 1;"
              >
                <div
                  class="leaflet-tile-container leaflet-zoom-animated"
                  style="z-index: 18; left: 0px; top: 0px;"
                >
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://maps.wikimedia.org/osm-intl/6/31/31.png"
                    style="width: 256px; height: 256px; left: -256px; top: -256px;"
                  />
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://maps.wikimedia.org/osm-intl/6/32/31.png"
                    style="width: 256px; height: 256px; left: 0px; top: -256px;"
                  />
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://maps.wikimedia.org/osm-intl/6/31/32.png"
                    style="width: 256px; height: 256px; left: -256px; top: 0px;"
                  />
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://maps.wikimedia.org/osm-intl/6/32/32.png"
                    style="width: 256px; height: 256px; left: 0px; top: 0px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="leaflet-pane leaflet-shadow-pane"
            >
              <img
                alt=""
                class="leaflet-marker-shadow leaflet-zoom-hide"
                src="marker-shadow.png"
                style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
              />
            </div>
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-marker-pane"
            >
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="marker-icon.png"
                style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
                tabindex="0"
              />
            </div>
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            />
            <div
              class="leaflet-top leaflet-right"
            />
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JS library for interactive maps"
                >
                  Leaflet
                </a>
                 | © 
                <a
                  href="http://osm.org/copyright"
                >
                  OpenStreetMap
                </a>
                 contributors
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-AxhUy laGaXi"
          data-testid="ClickIndicator"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-AxgMl eAPdQb"
            data-testid="SwipeIndicatorInnerUp"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 14l5-5 5 5z"
            />
          </svg>
        </div>
        <div
          class="sc-AxiKw boTYJG"
          data-testid="drawer"
        >
          <div
            class="sc-AxirZ evChcC"
            data-testid="handleBar"
          >
            <div
              class="sc-AxhCb gvnvGs"
              data-testid="swipeIndicator"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root sc-AxgMl eAPdQb"
                data-testid="SwipeIndicatorInner"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14l5-5 5 5z"
                />
              </svg>
            </div>
          </div>
          <div
            class="PrivateSwipeArea-root-5 PrivateSwipeArea-anchorBottom-9"
            style="height: 20px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="HomePage"
    >
      <div
        class="sc-fzoLag SLPgS"
        data-testid="TopBar"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root sc-fznyAO cWfjdE MuiIconButton-colorInherit MuiIconButton-edgeStart"
          data-testid="IconButtonStyled"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="sc-fzoXzr IxHii"
          data-testid="SearchFieldWrapper"
          role="button"
          tabindex="0"
        >
          <form>
            <div
              class="MuiFormControl-root MuiTextField-root sc-fzozJi hWuMcr sc-fzoLsD lgrigy"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  placeholder="search"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-2"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="sc-AxmLO fGDwgL"
        data-testid="CountriesList"
        style="opacity: 0; visibility: hidden;"
      >
        <div
          class="sc-Axmtr kQOpxN"
          data-testid="inlineLoader"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-fzpmMD bMrmBP leaflet-container leaflet-grab leaflet-touch-drag"
        style="position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; left: 0px; top: 0px;"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://maps.wikimedia.org/osm-intl/6/31/31.png"
                  style="width: 256px; height: 256px; left: -256px; top: -256px;"
                />
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://maps.wikimedia.org/osm-intl/6/32/31.png"
                  style="width: 256px; height: 256px; left: 0px; top: -256px;"
                />
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://maps.wikimedia.org/osm-intl/6/31/32.png"
                  style="width: 256px; height: 256px; left: -256px; top: 0px;"
                />
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://maps.wikimedia.org/osm-intl/6/32/32.png"
                  style="width: 256px; height: 256px; left: 0px; top: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-shadow-pane"
          >
            <img
              alt=""
              class="leaflet-marker-shadow leaflet-zoom-hide"
              src="marker-shadow.png"
              style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
            />
          </div>
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="marker-icon.png"
              style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
              tabindex="0"
            />
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          />
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
               | © 
              <a
                href="http://osm.org/copyright"
              >
                OpenStreetMap
              </a>
               contributors
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-AxhUy laGaXi"
        data-testid="ClickIndicator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root sc-AxgMl eAPdQb"
          data-testid="SwipeIndicatorInnerUp"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 14l5-5 5 5z"
          />
        </svg>
      </div>
      <div
        class="sc-AxiKw boTYJG"
        data-testid="drawer"
      >
        <div
          class="sc-AxirZ evChcC"
          data-testid="handleBar"
        >
          <div
            class="sc-AxhCb gvnvGs"
            data-testid="swipeIndicator"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root sc-AxgMl eAPdQb"
              data-testid="SwipeIndicatorInner"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 14l5-5 5 5z"
              />
            </svg>
          </div>
        </div>
        <div
          class="PrivateSwipeArea-root-5 PrivateSwipeArea-anchorBottom-9"
          style="height: 20px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
